### ISSUE TITLE
Error uploading research SV VCF

### ISSUE THREAD

**Issue by gavinmonahan** [Clinical-Genomics/scout#4078]
Hi all,

I'm getting an error with intervaltree while uploading the research SV VCF.
My pipeline is essentially to merge together multiple small cohorts of manta SV calls into one larger cohort using bcftools and truvari before annotating with VEP and genmod.
I have tried multiple iterations of the pipeline and it seems the issue appears after using [truvari collapse](https://github.com/acenglish/truvari/wiki/collapse), however I cannot see any issue with the VCF formatting that could cause this problem. I have also tested the VCF with multiple tools and validators and there were no errors. Uploading as a clinical SV VCF causes the same problem too. I'm using the docker-compose release and have successfully uploaded SV VCFs before.
The output of the upload command is:
```
worker@1a328a859f4c:~/app/scout/upload/configs$ scout --host mongodb load variants --sv 161
2023-09-19 06:49:59 1a328a859f4c scout.commands.base[3690] INFO Running scout version 4.70
2023-09-19 06:49:59 1a328a859f4c scout.commands.base[3690] DEBUG Debug logging enabled.
2023-09-19 06:49:59 1a328a859f4c scout.adapter.client[3690] INFO Connected to MongoDB 4.4.9
2023-09-19 06:49:59 1a328a859f4c scout.adapter.mongo.base[3690] DEBUG Database name: scout
2023-09-19 06:49:59 1a328a859f4c scout.server.app[3690] INFO Enable Phenopacket API
2023-09-19 06:49:59 1a328a859f4c scout.commands.load.variants[3690] INFO Running scout load variants
2023-09-19 06:49:59 1a328a859f4c scout.adapter.mongo.case[3690] INFO Nr variants with sanger verification found: 0
2023-09-19 06:49:59 1a328a859f4c scout.adapter.mongo.case[3690] INFO Nr variants with sanger ordered found: 0
2023-09-19 06:49:59 1a328a859f4c scout.commands.load.variants[3690] INFO Delete clinical sv variants for case 161
2023-09-19 06:49:59 1a328a859f4c scout.adapter.mongo.variant[3690] INFO Deleting old clinical sv variants for case 161
2023-09-19 06:49:59 1a328a859f4c scout.adapter.mongo.variant[3690] INFO 0 variants deleted
2023-09-19 06:49:59 1a328a859f4c scout.commands.load.variants[3690] INFO Load clinical sv variants for case 161
2023-09-19 06:49:59 1a328a859f4c scout.adapter.mongo.variant_loader[3690] DEBUG Attempt to load clinical SV VCF.
2023-09-19 06:49:59 1a328a859f4c scout.adapter.mongo.variant_loader[3690] INFO Found VEP header ALLELE|CONSEQUENCE|IMPACT|SYMBOL|GENE|FEATURE_TYPE|FEATURE|BIOTYPE|EXON|INTRON|HGVSC|HGVSP|CDNA_POSITION|CDS_POSITION|PROTEIN_POSITION|AMINO_ACIDS|CODONS|EXISTING_VARIATION|DISTANCE|STRAND|FLAGS|VARIANT_CLASS|SYMBOL_SOURCE|HGNC_ID|CANONICAL|MANE_SELECT|MANE_PLUS_CLINICAL|TSL|APPRIS|CCDS|ENSP|SWISSPROT|TREMBL|UNIPARC|UNIPROT_ISOFORM|GENE_PHENO|SIFT|POLYPHEN|DOMAINS|MIRNA|HGVS_OFFSET|AF|AFR_AF|AMR_AF|EAS_AF|EUR_AF|SAS_AF|GNOMADE_AF|GNOMADE_AFR_AF|GNOMADE_AMR_AF|GNOMADE_ASJ_AF|GNOMADE_EAS_AF|GNOMADE_FIN_AF|GNOMADE_NFE_AF|GNOMADE_OTH_AF|GNOMADE_SAS_AF|GNOMADG_AF|GNOMADG_AFR_AF|GNOMADG_AMI_AF|GNOMADG_AMR_AF|GNOMADG_ASJ_AF|GNOMADG_EAS_AF|GNOMADG_FIN_AF|GNOMADG_MID_AF|GNOMADG_NFE_AF|GNOMADG_OTH_AF|GNOMADG_SAS_AF|MAX_AF|MAX_AF_POPS|CLIN_SIG|SOMATIC|PHENO|PUBMED|OVERLAPBP|OVERLAPPC|MOTIF_NAME|MOTIF_POS|HIGH_INF_POS|MOTIF_SCORE_CHANGE|TRANSCRIPTION_FACTORS
2023-09-19 06:49:59 1a328a859f4c scout.adapter.mongo.hgnc[3690] INFO Fetching all genes
2023-09-19 06:50:00 1a328a859f4c scout.adapter.mongo.panel[3690] INFO Building gene to panels
2023-09-19 06:50:00 1a328a859f4c scout.adapter.mongo.hgnc[3690] INFO Building hgncid_to_gene
2023-09-19 06:50:00 1a328a859f4c scout.adapter.mongo.hgnc[3690] INFO Building interval trees...
2023-09-19 06:50:03 1a328a859f4c scout.adapter.mongo.variant_loader[3690] INFO Start inserting clinical sv variants into database
2023-09-19 06:50:03 1a328a859f4c scout.adapter.mongo.variant_loader[3690] ERROR unexpected error
Traceback (most recent call last):
  File "/venv/lib/python3.8/site-packages/intervaltree/intervaltree.py", line 1039, in __getitem__
    start, stop = index.start, index.stop
AttributeError: 'NoneType' object has no attribute 'start'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/worker/app/scout/adapter/mongo/variant_loader.py", line 687, in load_variants
    nr_inserted = self._load_variants(
  File "/home/worker/app/scout/adapter/mongo/variant_loader.py", line 415, in _load_variants
    managed = self._is_managed(variant, category)
  File "/home/worker/app/scout/adapter/mongo/variant_loader.py", line 565, in _is_managed
    coordinates = parse_coordinates(variant, category, build)
  File "/home/worker/app/scout/parse/variant/coordinates.py", line 198, in parse_coordinates
    "cytoband_end": get_cytoband_coordinates(end_chrom, end, build),
  File "/home/worker/app/scout/parse/variant/coordinates.py", line 27, in get_cytoband_coordinates
    for interval in coord_resource[chrom][pos]:
  File "/venv/lib/python3.8/site-packages/intervaltree/intervaltree.py", line 1048, in __getitem__
    return self.at(index)
  File "/venv/lib/python3.8/site-packages/intervaltree/intervaltree.py", line 800, in at
    return root.search_point(p, set())
  File "/venv/lib/python3.8/site-packages/intervaltree/node.py", line 297, in search_point
    if k.begin <= point < k.end:
TypeError: '<=' not supported between instances of 'int' and 'NoneType'
2023-09-19 06:50:03 1a328a859f4c scout.adapter.mongo.variant_loader[3690] WARNING Deleting inserted variants
2023-09-19 06:50:03 1a328a859f4c scout.adapter.mongo.variant[3690] INFO Deleting old clinical  variants for case 161
2023-09-19 06:50:03 1a328a859f4c scout.adapter.mongo.variant[3690] INFO 0 variants deleted
2023-09-19 06:50:03 1a328a859f4c scout.commands.load.variants[3690] WARNING '<=' not supported between instances of 'int' and 'NoneType'
```

I think this is likely an issue with my VCF, but any advice and help debugging this would be really appreciated, eg can we increase the verbosity of the error messages? It would be handy to know which part of the VCF is causing an error.

Thanks for making an awesome platform :)
Gavin

**Comment by northwestwitch** [comment]
Thanks Gavin! It looks like some lines of your VCF file are not in the expected format. Could you post the initial part of your VCF file, like the header plus a few variants, so we can check it?

**Comment by gavinmonahan** [comment]
Hi Chiara,
I have attached a few hundred lines here (and the header). Thanks for looking at this!
[truvari_vep_genmod_ranked_small.vcf.gz](https://github.com/Clinical-Genomics/scout/files/12657363/truvari_vep_genmod_ranked_small.vcf.gz)

**Comment by northwestwitch** [comment]
I found out what the issue is, it's when we compute the sv end coordinates, here: https://github.com/Clinical-Genomics/scout/blob/da36e2867485005cdcad7833713d3f47da4e9492/scout/parse/variant/coordinates.py#L60

You VCF has no "END" INFO field, but has a "SVLEN" INFO field that should be used in that function instead. This strategy is explained in the function's comment, but the code acts differently.

 I'll write a fix!

**Comment by northwestwitch** [comment]
Hello again @gavinmonahan , I wrote a fix in #4079. If you want you could check that it works by switching to that branch and trying to load the entire SV file from there!

**Comment by northwestwitch** [comment]
> Hey! I switched to that branch and it worked :) Thank you so much for the prompt fix

Thank you for reporting the error! 🥇
